attestation/factory.go

```diff
@@ -86,7 +86,20 @@ func FactoryByName(name string) (registry.FactoryFunc[Attestor], bool) {
 	return registrationEntry.Factory, ok
 }
 
+func AddAttestor(nameOrType string) (Attestor, error) {
+	attestors, err := AddAttestors([]string{nameOrType})
+	if err != nil {
+		return nil, err
+	}
+	return attestors[0], nil
```

Comment:
We may want to check that `len(attestors) > 0` prior to returning here to prevent a possible panic

Reply:
This has now prompted me to wonder. We currently use https://github.com/in-toto/go-witness/blob/404b65482adde0bdff021c20b6c49eab543d9e56/attestation/factory.go#L69-L71 if we can't find the attestor. Wouldn't this be better as a separate error saying the `Attestor` can't be found, rather than the `attestation`?